From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 23 Aug 2018 18:14:38 +0200
Subject: search: Cancel search provider operations on clear

Ensure that the search provider operations (just getResultMetas requests in the
current implementation) in progress are properly cancelled when we clear the UI,
otherwise returned results might still be added when not needed.

This is triggered for each provider by the SearchResults reset.

Bug-GNOME: https://gitlab.gnome.org/GNOME/gnome-shell/issues/183
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/bionic/+source/gnome-shell/+bug/1756826
Forwarded: https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/205
---
 js/ui/search.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/ui/search.js b/js/ui/search.js
index 1fb54b4..804be95 100644
--- a/js/ui/search.js
+++ b/js/ui/search.js
@@ -192,6 +192,7 @@ var SearchResultsBase = new Lang.Class({
     },
 
     clear() {
+        this._cancellable.cancel();
         for (let resultId in this._resultDisplays)
             this._resultDisplays[resultId].actor.destroy();
         this._resultDisplays = {};
